makefile

```diff
@@ -1,21 +1,25 @@
 NUM_NODES = 10
 WORLD_SIZE = 1000
 
-COMMON_CFLAGS = -g -std=gnu99 -O2 -mcpu=native -fomit-frame-pointer -Wall -Wextra
+COMMON_CFLAGS = -std=gnu99 -O2 -march=native -fomit-frame-pointer -Wall -Wextra
 
 buildall: c_fast c_fast_arm f03 c fsharp cpp_gcc cpp_clang cpp_cached racket csharp java haskell ocaml lisp rust rust_unsafe go gccgo d nim oraclejava crystal
 
 clean:
 	rm -f c_fast_arm c_fast f03 fs.exe cpp_gcc cpp_clang cpp_plain cpp_cached \
 		cs.exe jv.class hs ml lisp rs rs_unsafe go gccgo d nim crystal d \
 		c
+# C targets
+c: c.c
+	$(CC) $(COMMON_CFLAGS) c.c -o c -DUSE_HIGHBIT
 
 c_fast_arm: c_fast.c
-	gcc -marm -falign-functions=32 $(COMMON_CFLAGS) c_fast.c -o ./c_fast_arm
+	$(CC) -marm -falign-functions=32 $(COMMON_CFLAGS) c_fast.c -o ./c_fast_arm
 
 c_fast: c_fast.c
-	gcc -falign-functions=32 $(COMMON_CFLAGS) c_fast.c -o ./c_fast
+	$(CC) -falign-functions=32 $(COMMON_CFLAGS) c_fast.c -o ./c_fast
```

Comment:
The compiler was hardcoded to gcc on purpose, because I haven't evaluated or optimised clang's output for this implementation and therefore the code or flags might need some tweaks to achieve the optimum performance. I don't insist on keeping it that way, but maybe a comment would be in order.